src/briefcase/commands/base.py

```diff
@@ -117,24 +117,66 @@ class BaseCommand(ABC):
     def __init__(
         self,
         base_path,
-        home_path=Path.home(),
-        data_path=PlatformDirs(appname="briefcase", appauthor="BeeWare").user_data_path,
+        home_path=None,
+        data_path=None,
         apps=None,
         input_enabled=True,
     ):
+        # Some details about the host machine
+        self.host_arch = platform.machine()
+        self.host_os = platform.system()
+
         self.base_path = base_path
-        self.home_path = home_path
-        self.data_path = data_path
+
+        # If a home path is provided during construction, use it. This usually
+        # indicates we're under test conditions.
+        if home_path is None:
+            home_path = Path.home()
+        self.home_path = Path(home_path)
+
+        # If a data path is provided during construction, use it. This usually
+        # indicates we're under test conditions. If there's no data path
+        # provided, look for a BRIEFCASE_HOME environment variable. If that
+        # isn't defined, use a platform-specific default data path.
+        if data_path is None:
+            try:
+                data_path = os.environ["BRIEFCASE_HOME"]
```

Comment:
I think it might be good to ensure that whatever is in `BRIEFCASE_HOME` can be created as a directory or already exists as a directory via `mkdir(parents=True, exists_ok=True)`....otherwise, the bomb out later could be other than intuitive. (may even be good to see if `touch()` is successful but that may be overkill.)

Reply:
This was *partially* intentional; if the user specifies something incorrect, I don't want to start arbitrarily creating directories. However, verifying that the directory actually exists is a good idea.